src/pyhf/pdf.py

```diff
@@ -432,21 +462,27 @@ def expected_data(self, pars, include_auxdata=True):
             if expected_constraints is None
             else [expected_actual, expected_constraints]
         )
-        return tensorlib.concatenate(tocat)
+        if self.batch_size is None:
+            return tensorlib.concatenate(tocat, axis=0)
+        return tensorlib.concatenate(tocat, axis=1)
 
     def constraint_logpdf(self, auxdata, pars):
         tensorlib, _ = get_backend()
         normal = self.constraints_gaussian.logpdf(auxdata, pars)
         poisson = self.constraints_poisson.logpdf(auxdata, pars)
-        return tensorlib.sum(tensorlib.stack([normal, poisson]), axis=0)
+        if self.batch_size is None:
+            return normal + poisson
+        terms = tensorlib.stack([normal, poisson])
+        result = tensorlib.sum(terms, axis=0)
```

Comment:
This is style not substance, but could follow the return structure style of above and do

```suggestion
        return tensorlib.sum(terms, axis=0)
```

with removal of `result` below.